src/core/net/srp_client.hpp

```diff
@@ -842,13 +844,28 @@ class Client : public InstanceLocator, private NonCopyable
         OPENTHREAD_CONFIG_SRP_CLIENT_EARLY_LEASE_RENEW_FACTOR_DENOMINATOR;
 
     // -------------------------------
-    // When there is a change (e.g., a new service is added/removed)
-    // that requires an update, the SRP client will wait for a short
-    // delay as specified by `kUpdateTxDelay` before sending an SRP
-    // update to server. This allows the user to provide more change
-    // that are then all sent in same update message.
-    static constexpr uint32_t kUpdateTxMinDelay = OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_MIN_DELAY; // in msec.
-    static constexpr uint32_t kUpdateTxMaxDelay = OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_MAX_DELAY; // in msec.
+    // TX jitter constants
+    //
+    // When changes trigger a new SRP update message transmission a random
+    // jitter delay is applied before sending the update message to server.
+    // This can occur due to changes in client services or host info,
+    // or `AutoStart` selecting a server for first time or switching
+    // server thus requiring re-registration.
+    //
+    // The constants below specify jitter ranges applied based on
+    // different trigger reasons. All values are in milliseconds.
+    // Also see `TxJitter` class.
+
+    static constexpr uint32_t kMinTxJitter                  = 10;
+    static constexpr uint32_t kMaxTxJitterDefault           = 500;
+    static constexpr uint32_t kMaxTxJitterOnDeviceReboot    = 700;
+    static constexpr uint32_t kMaxTxJitterOnServerStart     = 10 * Time::kOneSecondInMsec;
+    static constexpr uint32_t kMaxTxJitterOnServerRestart   = 10 * Time::kOneSecondInMsec;
+    static constexpr uint32_t kMaxTxJitterOnServerSwitch    = 10 * Time::kOneSecondInMsec;
+    static constexpr uint32_t kMaxTxJitterOnSlaacAddrAdd    = 10 * Time::kOneSecondInMsec;
+    static constexpr uint32_t kMaxTxJitterOnSlaacAddrRemove = 10 * Time::kOneSecondInMsec;
+
+    static constexpr uint32_t kGuartTimeAfterAttachToUseShorterTxJitter = 1000;
```

Comment:
thanks for making the change. 

Is it possible we are handling TxJitter::kOnDeviceReboot, and select the short delay but immediately get the HandleUnicastAddressEvent, which would apply the longer jitter before the SRP update gets sent out the first time after reboot? 

Noticed minor correction (typo) to the variable name (kGuardTimeAfterAttachToUseShorterTxJitter)

Reply:
Thanks. 

> Is it possible we are handling TxJitter::kOnDeviceReboot, and select the short delay but immediately get the HandleUnicastAddressEvent, which would apply the longer jitter before the SRP update gets sent out the first time after reboot?

Yes, this is possible. In this case, `TxJitter` will track and apply the longer request.

It may be reasonable to apply the same `kGuardTime` for `kOnSlaacAddrAdd/Remove` events (i.e., if they occur shortly after the attach event, use a shorter delay). I'll explore adding this in the next push to this branch.

Reply:
Updated in new push. The `kGuardTime` is also checked for SLAAC address events.